File: lib/models/guardian.dart
class Guardian {
  final String id;
  final String userId;
  final String name;
  final String phone;
  final String relationship;
  final String? email;
  final int priority;
  final bool isVerified;
  final DateTime createdAt;
  final DateTime updatedAt;

  Guardian({
    required this.id,
    required this.userId,
    required this.name,
    required this.phone,
    required this.relationship,
    this.email,
    required this.priority,
    this.isVerified = false,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Guardian.fromJson(Map<String, dynamic> json) => Guardian(
    id: json['id'] as String,
    userId: json['userId'] as String,
    name: json['name'] as String,
    phone: json['phone'] as String,
    relationship: json['relationship'] as String,
    email: json['email'] as String?,
    priority: json['priority'] as int,
    isVerified: json['isVerified'] as bool? ?? false,
    createdAt: DateTime.parse(json['createdAt'] as String),
    updatedAt: DateTime.parse(json['updatedAt'] as String),
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'userId': userId,
    'name': name,
    'phone': phone,
    'relationship': relationship,
    'email': email,
    'priority': priority,
    'isVerified': isVerified,
    'createdAt': createdAt.toIso8601String(),
    'updatedAt': updatedAt.toIso8601String(),
  };

  Guardian copyWith({
    String? id,
    String? userId,
    String? name,
    String? phone,
    String? relationship,
    String? email,
    int? priority,
    bool? isVerified,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) => Guardian(
    id: id ?? this.id,
    userId: userId ?? this.userId,
    name: name ?? this.name,
    phone: phone ?? this.phone,
    relationship: relationship ?? this.relationship,
    email: email ?? this.email,
    priority: priority ?? this.priority,
    isVerified: isVerified ?? this.isVerified,
    createdAt: createdAt ?? this.createdAt,
    updatedAt: updatedAt ?? this.updatedAt,
  );
}

File: backend/routes/guardian.js
const express = require("express");
const Guardian = require("../models/Guardian");
const User = require("../models/User");
const authenticate = require("../middleware/authenticate");

const router = express.Router();

// Add guardian
router.post("/add", authenticate, async (req, res) => {
  try {
    const { name, phone, email, relationship, priority } = req.body;

    // Check if max guardians reached
    const user = await User.findById(req.userId);
    if (user.guardians.length >= 3) {
      return res.status(400).json({ error: "Maximum 3 guardians allowed" });
    }

    const guardian = new Guardian({
      userId: req.userId,
      name,
      phone,
      email,
      relationship,
      priority: priority || 1,
    });

    await guardian.save();

    // Add to user's guardians list
    user.guardians.push(guardian._id);
    await user.save();

    res.status(201).json({
      message: "Guardian added successfully",
      guardian,
    });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// Get all guardians
router.get("/list", authenticate, async (req, res) => {
  try {
    const guardians = await Guardian.find({ userId: req.userId }).sort({
      priority: 1,
    });
    res.json(guardians);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// Update guardian
router.put("/:id", authenticate, async (req, res) => {
  try {
    const { name, phone, email, relationship, priority } = req.body;

    const guardian = await Guardian.findByIdAndUpdate(
      req.params.id,
      {
        name: name || undefined,
        phone: phone || undefined,
        email: email || undefined,
        relationship: relationship || undefined,
        priority: priority || undefined,
        updatedAt: new Date(),
      },
      { new: true, runValidators: true }
    );

    if (!guardian) {
      return res.status(404).json({ error: "Guardian not found" });
    }

    res.json({
      message: "Guardian updated successfully",
      guardian,
    });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// Delete guardian
router.delete("/:id", authenticate, async (req, res) => {
  try {
    const guardian = await Guardian.findByIdAndDelete(req.params.id);

    if (!guardian) {
      return res.status(404).json({ error: "Guardian not found" });
    }

    // Remove from user's guardians list
    await User.findByIdAndUpdate(req.userId, {
      $pull: { guardians: req.params.id },
    });

    res.json({ message: "Guardian deleted successfully" });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

module.exports = router;
